src/sentry/static/sentry/app/components/stream/group.jsx

```diff
@@ -93,19 +105,65 @@ const StreamGroup = createReactClass({
     SelectedGroupStore.toggleSelect(this.state.data.id);
   },
 
+  toggleShowLifetimeStats(showLifetimeStats) {
+    if (this.hoverWait) {
+      clearTimeout(this.hoverWait);
+    }
+
+    this.hoverWait = setTimeout(
```

Comment:
I don't often review FE code here, so I may be a bit offbase. At my former company setTimeout was sign as an anti-pattern that was used as a last resort. Is this standard behavior here? I'd love to learn more since I didn't have a deep grasp as to why it would be good or bad.

Reply:
I don't have a strong opinion on this kind of meticulous ui behavior control. I saw this pattern in `Hovercard` which we use in a few places. I thought since this is a similar thing, I'd copy that pattern.

Reply:
Make sure to clearTimeout on hoverWait in componentWillUnmount as well to prevent errors after unmount.

Reply:
Could this also be an issue with the `Hovercard` component?